flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java

```diff
@@ -227,4 +235,28 @@ private boolean validateDeployment(FlinkResourceContext<FlinkDeployment> ctx) {
         }
         return true;
     }
+
+    private void setCRStatus(FlinkDeployment flinkApp) {
```

Comment:
Instead of having to call `setCrStatus` can we move this logic to within the actual `CommonStatus` or `FlinkDeploymentStatus` so that it's computed automatically like the `ResourceLifecycleState`?

Also I think we could actually compute this directly from the resource lifecycle state instead of adding arbitrary new logic here

Reply:
Made changes to populate Conditions in CommonStatus using resource lifecycle state